### PR TITLE
refactor: separate trie building from splicing leaf

### PR DESCRIPTION
**Pull request by rphmeier**

### CODE REVIEW

**Comment by rphmeier** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/thrumdev/nomt/141?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#141** <a href="https://app.graphite.dev/github/pr/thrumdev/nomt/141?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#140** <a href="https://app.graphite.dev/github/pr/thrumdev/nomt/140?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @rphmeier and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: core/src/update.rs [hunk 1]
```diff
@@ -188,19 +190,20 @@ pub struct VisitControl<'a> {
...
-// Provide a visitor which will be called for each computed node of the sub-trie.
+// Provide a visitor which will be called for each computed node of the trie.
 //
-// The visitor is assumed to have a default position at the root of the sub-trie and from
+// The visitor is assumed to have a default position at the root of the trie and from
 // there will be controlled with `VisitControl` to indicate the position of the next node.
 // The root is always visited at the end. If the written node is a leaf, the leaf-data preimage
 // will be provided.
-pub fn build_sub_trie<H: NodeHasher>(
+pub fn build_trie<H: NodeHasher>(
```
Comment (pepyakin) [inline core/src/update.rs:204]: I think at this point the doc above requires overhaul.

**Review by pepyakin — APPROVED** [review]
Looks good, but tbh I don't have such a good insight of this part as you do and it's not clear for me what would this enables or fixes and as such it would have beeen awesome if you left a short description.

**Comment by pepyakin** [comment]
### Merge activity

* **Apr 18, 2:53 AM EDT**: @pepyakin started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/thrumdev/nomt/141).

**Pull request merged**